internal/api/user_devices.go

```diff
@@ -267,6 +267,37 @@ func (s *userDeviceService) RegisterUserDeviceFromVIN(ctx context.Context, req *
 	return &pb.RegisterUserDeviceFromVINResponse{Created: true}, err
 }
 
+func (s *userDeviceService) UpdateDeviceIntegrationStatus(ctx context.Context, req *pb.UpdateDeviceIntegrationStatusRequest) (*pb.UserDevice, error) {
+	dbDevice, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(req.UserDeviceId),
```

Comment:
the `user_device_api_integration` also has the column `user_device_id` so you may be able to simplify and only select from `user_device_api_integration`